File: ruby/two_sum.rb

def two_sum1(nums, target)
  nums.each_with_index do |num, index|
    leftNums = nums.slice(index + 1, nums.size - index - 1) if nums.size - index -1 > 0
    return [index, leftNums.index(target - num) + index + 1] if leftNums&.index(target - num)
  end
  return []
end

def two_sum2(nums, target)
  lookfor = {}
  nums.each_with_index do |num, index| 
    return [lookfor[num], index] if lookfor[num]
    lookfor[target - num] = index
  end
end

File: js/two_sum.js
function two_sum(nums, target){
	let pair = {};
	let answer = [];
	for(let index in nums){
		let num = nums[index];
		let padding = target - num;
		if(pair[num]){
			answer = [pair[num], index];
			break;
		}else{
			pair[padding] = index;
		}
	}
	return answer;
}

console.log(two_sum([2, 7, 11, 15], 9));
